### DISCUSSION TITLE
Timeout when connecting to XMPP server

### DISCUSSION THREAD

**Discussion by nioc** [nioc/xmpp-web#75] [Ideas]
As discussed with @Ppjet6  on support@joinjabber.org, sometimes user encounters a "Server unreachable" message when connecting.
For now timeout is set to 5 seconds, but maybe it is too short?

**Comment by nioc** [comment]
I propose to add this timeout as local.js configuration, but may be `@xmpp/client` library [use its own timeout](https://github.com/xmppjs/xmpp.js/blob/main/packages/connection/index.js#L16) (set to 2 seconds)... We have to test 😉

**Comment by Ppjet6** [comment]
I use Tor, so this can explain the latency/slowness.

The thing though is that most of the time the connection is actually made to the server (looking at the websocket request), it's just slow. It seems to stop after disco#items on the host. And the interface doesn't pick it up. It just displays "Server unreachable" way too fast.
